### PR TITLE
feat(FX-3634): Allow to update search criteria attributes from saved alerts screen

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feature**

This PR resolves [FX-3634]

### Description
Currently, when making changes in a saved alert, we can update only name, email and push. We need to make changes and add the ability to update search criteria attributes on the edit saved search alert screen

### Acceptance Criteria
* As a collector
* When I’m on the edit alert screen
* And I remove a non-default filter
* And I click Save Alert
* Then my alert is updated to the desired criteria

##### iOS
https://user-images.githubusercontent.com/3513494/150136271-b7d6e1f7-8388-4cbc-b0f9-cbf3875fd9d8.mp4

##### Android
https://user-images.githubusercontent.com/3513494/150138195-08e953e0-a4eb-4a38-a57c-a6a47c53ae2e.mp4

####  A user tries to update an alert to match a pre-existing one
https://user-images.githubusercontent.com/3513494/150136496-f0083b45-099e-4c18-bc21-92ac71748941.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [x] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Allow to update search criteria attributes from saved alerts screen - dimatretyak

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[FX-3634]: https://artsyproduct.atlassian.net/browse/FX-3634?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Allow to update search criteria attributes from saved alerts screen - dimatretyak)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2f4b67abd2a69098ab3e98dca8edb73a4eed7c5f
</p>

File: src/lib/Scenes/SavedSearchAlert/mutations/updateSavedSearchAlert.ts [hunk 1]
```diff
@@ -1,15 +1,28 @@
...
+import { SearchCriteriaAttributes } from "lib/Components/ArtworkFilter/SavedSearch/types"
 import { defaultEnvironment } from "lib/relay/createEnvironment"
 import { commitMutation, graphql } from "relay-runtime"
 import { SavedSearchAlertFormValues } from "../SavedSearchAlertModel"
 
 export const updateSavedSearchAlert = (
+  savedSearchAlertId: string,
   userAlertSettings: SavedSearchAlertFormValues,
-  savedSearchAlertId: string
+  attributes: SearchCriteriaAttributes | null
```
Comment (gkartalis) [inline src/lib/Scenes/SavedSearchAlert/mutations/updateSavedSearchAlert.ts:14]: **question:** Shouldn't this be like
```suggestion
  attributes?: SearchCriteriaAttributes
```

**Review by gkartalis — APPROVED** [review]
👍  Looking great, just added a question

File: src/lib/Scenes/SavedSearchAlert/mutations/updateSavedSearchAlert.ts (hunk 1 above)
Comment (dimatretyak) [reply src/lib/Scenes/SavedSearchAlert/mutations/updateSavedSearchAlert.ts:14]: Yes, a good suggestion. I'll add it now

**Review by gkartalis — APPROVED** [review]

**Pull request merged**